src/cartservice/cartstore/InstrumentedCartStore.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Threading.Tasks;
+using cartservice.interfaces;
+using Hipstershop;
+using OpenCensus.Trace;
+using OpenCensus.Trace.Sampler;
+
+namespace cartservice.cartstore
+{
+    /// <summary>
+    /// Wrapper for Cart Store - instrumented with OpenCensus tracing
+    /// </summary>
+    internal class InstrumentedCartStore : ICartStore
+    {
+        private readonly ICartStore cartStore;
+        private static ITracer tracer = Tracing.Tracer;
+        private ISpanBuilder initializeSpanBuilder, addItemSpanBuilder, emptySpanBuilder, getItemSpanBuilder, pingSpanBuilder; 
+
+        public InstrumentedCartStore(ICartStore cartStore)
+        {
+            this.cartStore = cartStore;
+
+            // Create Span Builders for tracing
+            initializeSpanBuilder = CreateSpanBuilder("Initialize Cart Store");
```

Comment:
can't we infer span names from RPCs automatically? (all other services do that) If so, please stick to RPC names for the sake of consistency/searchability.

we shouldn't be guessing what's the span name for things.

also, most other languages don't create a span –spans are created by gRPC OC wrapper automatically. Don't we have that in C# yet?

Reply:
We don't have grpc native instrumentation in C# yet. I am planning to add that in the next 1.5 months. Even if we would have had that, it only automatically instruments remote procedure calls. If we want to instrument logical operations in cart service itself, or calls to third party libraries that are not natively instrumented by opencensus, we must instrument ourselves. Other services in Hipstershop don't work with external services so much, so they might not need this.

Reply:
I think span titles for internal operations or external calls are fine, but I assume we should hold off on to this PR until C# can use HTTP request paths from GRPC as span names?

I'm happy to merge this if you say so, but you'll need to refactor once it's available.